### PR TITLE
Build scripts refactoring

### PR DESCRIPTION
**Pull request by zstadler**
`.github/build-and-upload.sh` to use `build.sh`
Related to  #32

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -5,16 +5,26 @@ cat <<USAGE
...
+  ./build.sh [[--push] <tag>]
+  ./build.sh --help
 
 Argument:
-  <tag>   Build an image for the given graphhopper repository tag [default: master]
+  <tag>         Build an image for the given graphhopper repository tag [default: master]
 
 Option:
-  --help  Print this message
+  --push        Push the image to Docker Hub
```
Comment (HarelM) [inline build.sh:15]: Add explanation regarding possible usage of environment variables for login

File: build.sh [hunk 2]
```diff
@@ -32,8 +45,11 @@ if [ "$1" ]; then
...
 
 echo "Building docker image ${imagename}"
-docker build . -t ${imagename}
+docker build . -t "${imagename}"
 
-if [ $# -eq 1 ]; then
+if [ "${push}" == "false" ]; then
   echo "Use \"docker push ${imagename}\" to publish the image on Docker Hub"
+else
+  # TODO
```
Comment (HarelM) [inline build.sh:53]: Is there a TODO left here? This seems fine to me...

**Comment by HarelM** [comment]
This looks great, thanks! See my minor comments.

File: build.sh (hunk 1 above)
Comment (zstadler) [reply build.sh:15]: Done in a4cbdd2e0d9602921d833807f40ec3bfc4fba066

File: build.sh (hunk 2 above)
Comment (zstadler) [reply build.sh]: Comment removed in a4cbdd2e0d9602921d833807f40ec3bfc4fba066

**Review by HarelM — APPROVED** [review]

**Pull request merged**